File: backend/src/main/java/de/interaapps/pastefy/model/database/Notification.java
package de.interaapps.pastefy.model.database;

import org.javawebstack.orm.Model;
import org.javawebstack.orm.annotation.Column;
import org.javawebstack.orm.annotation.Dates;
import org.javawebstack.orm.annotation.Table;

import java.sql.Timestamp;

@Dates
@Table("notification")
public class Notification extends Model {
    @Column
    public int id;

    @Column
    private String message;

    @Column
    public int userId;

    @Column
    public String url;

    @Column
    public boolean alreadyRead = false;

    @Column
    public boolean received = false;


    @Column
    public Timestamp createdAt;

    @Column
    public Timestamp updatedAt;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public int getId() {
        return id;
    }

    public int getUserId() {
        return userId;
    }

    public String getUrl() {
        return url;
    }

    public void setAlreadyRead(boolean alreadyRead) {
        this.alreadyRead = alreadyRead;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setReceived(boolean received) {
        this.received = received;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }
}

File: backend-kt/src/main/kotlin/de/interaapps/pastefy/entities/Notification.kt
package de.interaapps.pastefy.entities

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(
    name = "pastefy_notification",
    indexes = [
        Index(name = "pastefy_notification_user_id_received_index", columnList = "user_id, received"),
    ],
)
class Notification(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null,
    var message: String? = null,
    @Column(length = 8, nullable = false) var userId: String = "",
    var url: String? = null,
    @Column(nullable = false) var alreadyRead: Boolean = false,
    @Column(nullable = false) var received: Boolean = false,
    @Column(nullable = false, updatable = false) var createdAt: Instant? = null,
    @Column(nullable = false) var updatedAt: Instant? = null,
) {
    @PrePersist fun prePersist() {
        val now = Instant.now()
        if (createdAt == null) createdAt = now
        updatedAt = now
    }
    @PreUpdate fun preUpdate() { updatedAt = Instant.now() }
}
